### PR TITLE
Corgis will no longer pick up non-items (like entire cars)

### LINKED ISSUE #59202: Corgis can fetch any pullable object
## Round ID:

[Round ID]: N/A

## Reproduction:
Reported on Fulpstation running on TG PR https://github.com/tgstation/tgstation/pull/58822 as latest.

Corgis can fetch anything that can be pulled like lockers and IV drips, and ignore directional windows, so they can fetch through walls as long as the item is adjacent.
This is both hilarious and game-breaking once you realize Ian can get a clown car in his mouth and make everyone in the car suffocate because of it

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
I made one hell of a whoopsie with dog AI, letting dogs fetch and pick up and carry around things that are absolutely not meant to be carryable, basically anything that isn't anchored. This restricts their carrying ability to items.

Fixes: #59202

## Why It's Good For The Game
Dogs should not be able to hold cars in their mouths

## Changelog
:cl: Ryll/Shaps
fix: Dogs are now properly restricted to being able to fetch/pick up items only, and can no longer fetch you lockers and full size cars
/:cl:

### CODE REVIEW

**Comment by maxymax13** [comment]
no more clown car vore 🤕

File: code/datums/ai/dog/dog_controller.dm [hunk 1]
```diff
@@ -301,21 +304,22 @@
...
 		return
-	var/mob/living/living_pawn = pawn
-	if(IS_DEAD_OR_INCAP(living_pawn))
-		return
 
 	COOLDOWN_START(src, command_cooldown, AI_DOG_COMMAND_COOLDOWN)
 
 	switch(blackboard[BB_DOG_ORDER_MODE])
 		if(DOG_COMMAND_FETCH)
-			if(ismob(pointed_movable) || pointed_movable.anchored)
```
Comment (ninjanomnom) [inline code/datums/ai/dog/dog_controller.dm:318]: Check the item weight class as well, at the very least disallowing `WEIGHT_CLASS_GIGANTIC` and probably `WEIGHT_CLASS_HUGE` as well

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/ai/dog/dog_controller.dm (hunk 1 above)
Comment (maxymax13) [reply code/datums/ai/dog/dog_controller.dm:318]: WEIGHT_CLASS_HUGE means no fetching shotguns :(

Comment (AnturK) [reply code/datums/ai/dog/dog_controller.dm:318]: Weight class checks seem kinda pointless to me but should probably check for ABSTRACT flag.

**Pull request merged**